Creation of audit and aws schema

LucioDonda opened the pull request:
|Related issue|
|---|
|#16322|

<!--
This template reflects sections that must be included in new Pull requests.
Contributions from the community are really appreciated. If this is the case, please add the
"contribution" to properly track the Pull Request.

Please fill the table above. Feel free to extend it at your convenience.
-->

## Description
In order to use the new schema script in this PR the schemas needed for AWS and audited decoders are added

## Task
- [x] Create audit scheme
- [x] Create AWS schema
- [x] Update Documentation
- [x] QA validate

JcabreraC commented:
Beware, the archives:
- src/engine/extension/schema/aws/fields.yml
- src/engine/extension/elasticsearch/7.x/wazuh-template.json
- src/engine/ruleset/schemas/wazuh-logpar-types.json

say they have completely changed because the tabulation has been changed. Change it to show only the PR changes.

JcabreraC reviewed (changes requested):
The `src/engine/extension/schema/auditd/fields.yml` file is missing fields that decode the decoders.

LucioDonda commented:
Auditd fields analysis:

- Fields used as temporary inside of the decoder:

auditd.log

- Fields renamed inside of the decoder:

auditd.log.acct
auditd.log.arch
auditd.log.comm
auditd.log.cwd
auditd.log.exe

- Fields saved in another:

auditd.log.cmd
auditd.log.src
auditd.log.record_type

- Deleted at the end of decoders:

auditd.log.copy
auditd.log.epoch
auditd.log.sub_kv
auditd.log.kv


- Added fields (moved from `.log.` ):

auditd.data.apparmor
auditd.data.cgroup
auditd.data.file
auditd.data.device
auditd.data.dir
auditd.data.grp
auditd.data.invalid_context
auditd.data.new-chardev
auditd.data.new-disk
auditd.data.new-fs
auditd.data.new-net
auditd.data.new-rng
auditd.data.ocomm
auditd.data.old-chardev
auditd.data.old-disk
auditd.data.old-fs
auditd.data.old-net
auditd.data.old-rng
auditd.data.success
auditd.data.vm
auditd.data.watch

- New fields (taken from hex script) :

auditd.log.node -> ~auditd.log.node
auditd.log.denied_mask
auditd.log.new_group
auditd.log.info
auditd.log.operation
auditd.log.proctitle
auditd.log.profile
auditd.log.requested_mask
auditd.log.root_dir
auditd.log.sw
auditd.log.data
auditd.log.path

nmkoremblum commented on src/engine/ruleset/decoders/auditd/auditd-kv.yml:
```diff
@@ -26,7 +26,7 @@ parse:
   - map:
       - auditd.log: +ef_merge/$~kv_map
-      - auditd.log.node: +ef_rename/$~node
+      - ~auditd.log.node: +ef_rename/$~node
```
Why has this field been changed to a temporal one?

LucioDonda replied:
Going back to the pipeline, taken as a base for the decoder, that field is used and mapped as in this case, but I couldn't find any description of it or any other use in the schemas. IMHO It wasn't a wrong approach to define it as temporary, but if it is needed after, we could create a custom field with that name.

nmkoremblum reviewed (approved):
LGTM :+1:

juliancnn reviewed (approved):
LGTM!, only just need to fix sophos
```
    engine: "/home/vagrant/engine/wazuh/src/engine/ruleset/decoders/sonicwall/sonicwall-firewall-generated.yml" ==> An error occurred while trying to validate "decoder/sonicwall-firewall-generated/0": Parse stage: Building asset "decoder/sonicwall-firewall-generated/0" failed: Error building parser "logpar": An error occurred while parsing a log: Field 'observer.egress.interface.name' not found in schema
```

nbertoldo reviewed (approved):
LGTM!

JcabreraC reviewed (approved):
LGTM !
